app/models/labware_creators/plate_split_to_tube_racks.rb

```diff
@@ -540,45 +576,105 @@ def find_child_tube(well)
       end
     end
 
-    # Adds metadata to child tubes using details from the parsed sequencing and contingency CSV files.
+    # Adds metadata to child tubes using details from the parsed sequencing and
+    # contingency CSV files, and the Cellaca cell count information from the parent plate.
     #
     # @return [void]
-    def add_child_tube_metadata
-      add_sequencing_tube_metadata unless require_contingency_tubes_only?
-
-      add_contingency_tube_metadata
+    def update_child_tube_metadata
+      tubes_metadata = {}
+      add_sequencing_tube_metadata(tubes_metadata) unless require_contingency_tubes_only?
+      add_contingency_tube_metadata(tubes_metadata)
+      add_cell_count_metadata(tubes_metadata)
+      write_metadata_to_tubes(tubes_metadata)
     end
 
     # Adds tube rack barcode and position metadata to child sequencing tubes.
     #
     # @return [void]
-    def add_sequencing_tube_metadata
+    def add_sequencing_tube_metadata(tubes_metadata)
```

Comment:
Code climate had a point 😆 there is some code duplication here. Maybe GitHub copilot can refactor it for you?